src/map/unit.cpp

```diff
@@ -3260,13 +3273,8 @@ static int32 unit_attack_timer_sub(struct block_list* src, int32 tid, t_tick tic
 			ud->skill_id = 0;
 
 		// You can't move during your attack motion
-		if (src->type&battle_config.attack_walk_delay) {
-			// Monsters set their AI inactive instead of having a walkdelay
-			if (md != nullptr)
-				md->next_thinktime = tick + status_get_amotion(src);
-			else
-				unit_set_walkdelay(src, tick, sstatus->amotion, 1);
-		}
+		if (src->type&battle_config.attack_walk_delay)
+			unit_set_walkdelay(src, tick, sstatus->amotion, 1);
```

Comment:
Why dont we do that in unit_set_attackdelay?

Reply:
Originally I was thinking about integrating unit_set_walkdelay into unit_set_attackdelay later on, but since I know now that this is just a custom feature, I kinda want to leave it separate.

Reply:
battle_config.mob_ai&0x2000 is also a custom feature and was also moved into the function